internal/errs/unwrap.go

```diff
@@ -6,13 +6,29 @@ import (
 	"github.com/ActiveState/cli/internal/failures"
 )
 
+func unwrapExitError(err error) (bool, int) {
+	var eerr interface{ ExitCode() int }
+	isExitError := errors.As(err, &eerr)
+	if isExitError {
+		// If exit error happened in activated shell, do not forward exit code, but return 0
+		var activatedErr interface{ IsFromActivatedShell() }
+		isFromActivatedShell := errors.As(err, &activatedErr)
+		if isFromActivatedShell {
+			return true, 0
+		}
```

Comment:
This chunk feels unnecessary. We already know it's an exit error. We don't need to know whether it came from an activated state. This just seems to override the exitCode, which shouldn't be necessary; the activate runner should convey this. eg. wrap it in an ExitCode=0 error.

Reply:
That is a good idea.  If you haven't yet, please also consider my comment to the PR: https://www.pivotaltracker.com/story/show/175651734/comments/219641816

After implementing this, I believe that it would be better not to change the error code to 0, but to just silent the error message.  After all, that was what the complaint during the demo was about, and changing the error code changes the scripts behavior.  I am especially concerned about the `state activate -c` use case, that we have in the visual studio code extension.

Reply:
> 
> 
> Please move the error handling changes (failures > error) into a separate PR. Normally I'd be welcoming of this type of change provided it's not too big, which in this case I don't feel qualifies.

I actually wanted to explain myself here, before I requested a review: I found these changes warranted, because the exit code handling that was duplicated for failures and errors complicated things quite a bit.  So, I was happy that I could move it all to failures.

Reply:
> That is a good idea. If you haven't yet, please also consider my comment to the PR: https://www.pivotaltracker.com/story/show/175651734/comments/219641816
> 
> After implementing this, I believe that it would be better not to change the error code to 0, but to just silent the error message. After all, that was what the complaint during the demo was about, and changing the error code changes the scripts behavior. I am especially concerned about the `state activate -c` use case, that we have in the visual studio code extension.

I'm on board with this. Agree forwarding the exit code isn't really the issue here, it's the output we provide around it.

Reply:
> I actually wanted to explain myself here, before I requested a review: I found these changes warranted, because the exit code handling that was duplicated for failures and errors complicated things quite a bit. So, I was happy that I could move it all to errors.
> 
> Please let me know how you decide on the PT story comments, and I will implement it without the part that removes the failure handling.

Ok. Let's go with it then, but please tripple check that you didn't miss any failures. Be mindful also of functions that called your function that now returns an error rather than a failure.